collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -179,6 +184,25 @@ public void logFormEvent(String event) {
         return hasBackgroundRecording;
     }
 
+    public FormEntryPrompt getQuestionPrompt(FormIndex formIndex) {
+        return formController.getQuestionPrompt(formIndex);
+    }
+
+    public void setAnswerListener(@Nullable AnswerListener answerListener) {
+        this.answerListener = answerListener;
+    }
+
+    public void answerQuestion(FormIndex index, IAnswerData answer) {
```

Comment:
I wanted to avoid using the `WaitingForDataRegistry` for this as it felt too heavy weight. This solution feels much better to me - it currently wouldn't support the acivity result use cases but should work nicely for cases that are "internal" to our app. It's currently using a listener rather than a reactive value pattern (`LiveData`) as I didn't want to move to introduce the storing answer data in a `ViewModel` quite yet (I think eventually we'd want to get to that though).